From 28a8bd4b10480083aaea40b062d771e3d135ce03 Mon Sep 17 00:00:00 2001
From: Pankaj Chauhan <pankaj.chauhan@freescale.com>
Date: Tue, 23 Dec 2014 17:20:52 +0530
Subject: [PATCH 0119/1429] irq/msi: Hack to avoid IRQ creation failure

get_hwirq() returns a uninitialized hwirq value because
hwirq is allocated in irq_domain_alloc_irqs_parent().

Moreover GIC ITS driver doesn't update hwirq in msi_alloc_info.hwirq
(arg points to msi_alloc_info), so taking out hwirq using get_hwirq
will anyways fail even if get_hwirq() is called after
irq_domain_alloc_irqs_parent().

Signed-off-by: Pankaj Chauhan <pankaj.chauhan@freescale.com>
Change-Id: I94f98fc3f605d91ba654bbe3059ae152dff61db2
Reviewed-on: http://git.am.freescale.net:8181/33324
Reviewed-by: Stuart Yoder <stuart.yoder@freescale.com>
Tested-by: Stuart Yoder <stuart.yoder@freescale.com>
(cherry picked from commit c9ea883c17729bb93aa55b08840ae48c8b495142)
[Original patch taken from QorIQ-SDK-V2.0-20160527-yocto]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 kernel/irq/msi.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/kernel/irq/msi.c b/kernel/irq/msi.c
index 474de5c..49e1b80 100644
--- a/kernel/irq/msi.c
+++ b/kernel/irq/msi.c
@@ -89,8 +89,10 @@ static int msi_domain_alloc(struct irq_domain *domain, unsigned int virq,
 	irq_hw_number_t hwirq = ops->get_hwirq(info, arg);
 	int i, ret;
 
+#if 0
 	if (irq_find_mapping(domain, hwirq) > 0)
 		return -EEXIST;
+#endif
 
 	ret = irq_domain_alloc_irqs_parent(domain, virq, nr_irqs, arg);
 	if (ret < 0)
-- 
1.7.5.4

